beacon-chain/db/kv/migration_state_validators.go

```diff
@@ -0,0 +1,91 @@
+package kv
+
+import (
+	"bytes"
+	"context"
+
+	"github.com/golang/snappy"
+	v1 "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+	"github.com/prysmaticlabs/prysm/shared/featureconfig"
+	"github.com/prysmaticlabs/prysm/shared/hashutil"
+	bolt "go.etcd.io/bbolt"
+)
+
+var migrationStateValidators0Key = []byte("migration_state_validator")
+
+func migrateStateValidators(tx *bolt.Tx) error {
+	if !featureconfig.Get().EnableHistoricalSpaceRepresentation {
+		return nil
+	}
+	mb := tx.Bucket(migrationsBucket)
+	if b := mb.Get(migrationStateValidators0Key); bytes.Equal(b, migrationCompleted) {
+		return nil // Migration already completed.
+	}
+
+	// get the source and destination buckets.
+	log.Infof("migrating %s. It will take few minutes. Please wait.", stateBucket)
```

Comment:
Highly recommended to use progressutil here from our shared package. If it will take a few minutes, it is imperative the user has feedback in the meantime
```suggestion
	log.Infof("Performing a one-time migration to a more efficient database schema for %s. It will take few minutes", stateBucket)
```